src/mpl/src/gpu/mpl_gpu_hip.c

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#include "mpl.h"
+
+MPL_SUPPRESS_OSX_HAS_NO_SYMBOLS_WARNING;
+
+#ifdef MPL_HAVE_HIP
+
+#define HIP_ERR_CHECK(ret) if (unlikely((ret) != hipSuccess)) goto fn_fail
+
+int MPL_gpu_query_pointer_type(const void *ptr, MPL_pointer_type_t * attr)
+{
+    hipError_t ret;
+    hipPointerAttributes_t ptr_attr = 0;
+    ret = hipPointerGetAttributes(&ptr_attr, ptr);
+    HIP_ERR_CHECK(ret);
+    switch (ptr_attr.type) {
+        case hipMemoryTypeUnregistered:
+        case hipMemoryTypeHost:
+            *attr = MPL_GPU_POINTER_HOST;
+            break;
+        case hipMemoryTypeDevice:
+            *attr = MPL_GPU_POINTER_DEV;
+            break;
+        case hipMemoryTypeManaged:
+            *attr = MPL_GPU_POINTER_MANAGED;
+            break;
+    }
+
+  fn_exit:
+    return MPL_GPU_SUCCESS;
+  fn_fail:
+    return MPL_GPU_FAILURE;
+}
+
+int MPL_gpu_ipc_get_mem_handle(MPL_gpu_ipc_mem_handle_t * h_mem, void *ptr)
+{
+    MPL_Abort();
```

Comment:
MPL functions use a lower-case character after `_`.  Also, if we are simply going to abort, why are we adding HIP support?  It might be better to delete it.

Reply:
I guess we should probably remove this as it is of little use until Yaksa support for AMD is available.

Reply:
Removed.